src/index.js

```diff
@@ -1,15 +1,17 @@
-import { useEffect, useRef } from "react";
-import io from "socket.io-client";
+import { useEffect, useState } from 'react';
+import io from 'socket.io-client';
+
+export const useSocket = (...args) => {
+  const [socket, setSocket] = useState(null);
 
-const useSocket = (...args) => {
-  const { current: socket } = useRef(io(...args));
   useEffect(() => {
+    setSocket(() => io(...args));
```

Comment:
why don't you set instance?
```suggestion
    setSocket(io(...args));
```

Reply:
Not sure why I did that. I tried many approaches so maybe I tried something and forgot to remove it.